opentelemetry-sdk/src/trace/tracer.rs

```diff
@@ -95,27 +95,43 @@ impl Tracer {
         &self,
         sampling_result: SamplingResult,
         parent_cx: &Context,
-    ) -> Option<(TraceFlags, Vec<KeyValue>, TraceState)> {
+    ) -> (SamplingDecision, TraceFlags, Vec<KeyValue>, TraceState) {
         match sampling_result {
             SamplingResult {
                 decision: SamplingDecision::Drop,
+                trace_state,
                 ..
-            } => None,
+            } => (
+                SamplingDecision::Drop,
+                TraceFlags::default(),
+                Vec::new(),
```

Comment:
The SDK should do minimum processing in case the spans are dropped. Do you think we can replace it with an empty slice, which can be statically allocated?

Reply:
According to @TommyCpp , `process_sampling_result ` can be deprecated